code/modules/clothing/accessories/holster.dm

```diff
@@ -123,8 +123,8 @@
 	desc = "A handgun holster that clips to a suit. Perfect for concealed carry."
 	holster_verb_name = "Holster (Handgun)"
 
-/obj/item/clothing/accessory/holster/handgun/can_holster(obj/item/weapon/gun/W)
-	if(!istype(W))
+/obj/item/clothing/accessory/holster/handgun/can_holster(obj/item/weapon/W)
+	if(!isgun(W) && !isbanana(W) && !istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/bluespacebanana))
```

Comment:
because a bluespace banana isn't a banana!

Reply:
You were the one who suggested it should be a child, this seems like the ideal use, a function to treat all bananas alike even if they are not inheriting behaviors

Reply:
But what if you want to do something banana-specific and you use isbanana but it breaks on bluspace bananas because bluespace bananas aren't actually bananas

Reply:
you would just istype the path of banana. Why are you making a macro for an operation performed exactly one time anyway?